code/modules/admin/stickyban.dm

```diff
@@ -0,0 +1,211 @@
+				return
+
+			var/found = 0
+			//we have to do it this way because byond keeps the case in its sticky ban matches WHY!!!
+			for (var/key in ban["keys"])
+				if (ckey(key) == alt)
+					found = 1
+					break
+
+			if (!found)
+				usr << "<span class='adminnotice'>Error: [alt] is not linked to [ckey]'s sticky ban!</span>"
+				return
+
+			if (alert("Are you sure you want to disassociate [alt] from [ckey]'s sticky ban? \nNote: Nothing stops byond from re-linking them","Are you sure","Yes","No") == "No")
+				return
+
+			//we have to do this again incase something changes
+			ban = get_stickyban_from_ckey(ckey)
+			if (!ban)
+				usr << "<span class='adminnotice'>Error: The ban disappeared.</span>"
+				return
+
+			found = 0
+			for (var/key in ban["keys"])
+				if (ckey(key) == alt)
+					ban["keys"] -= key
+					found = 1
+					break
+
+			if (!found)
+				usr << "<span class='adminnotice'>Error: [alt] link to [ckey]'s sticky ban disappeared.</span>"
+				return
+
+			world.SetConfig("ban",ckey,list2stickyban(ban))
+
+			log_admin("[key_name(usr)] has disassociated [alt] from [ckey]'s sticky ban")
+			message_admins("<span class='adminnotice'>[key_name_admin(usr)] has disassociated [alt] from [ckey]'s sticky ban</span>")
+
+		if ("edit")
+			if (!data["ckey"])
+				return
+			var/ckey = data["ckey"]
+			var/ban = get_stickyban_from_ckey(ckey)
+			if (!ban)
+				usr << "<span class='adminnotice'>Error: No sticky ban for [ckey] found!"
+				return
+			var/oldreason = ban["message"]
+			var/reason = input(usr,"Reason","Reason","[ban["message"]]") as text|null
+			if (!reason || reason == oldreason)
+				return
+			//we have to do this again incase something changed while we waited for input
+			ban = get_stickyban_from_ckey(ckey)
+			if (!ban)
+				usr << "<span class='adminnotice'>Error: The ban disappeared.</span>"
+				return
+			ban["message"] = "[reason]"
+
+			world.SetConfig("ban",ckey,list2stickyban(ban))
+
+			log_admin("[key_name(usr)] has edited [ckey]'s sticky ban reason from [oldreason] to [reason]")
+			message_admins("<span class='adminnotice'>[key_name_admin(usr)] has edited [ckey]'s sticky ban reason from [oldreason] to [reason]</span>")
+	
+	spawn(10)
+		stickyban_show()
+
+/datum/admins/proc/stickyban_gethtml(ckey, ban)
+	. = "<a href='?_src_=holder;stickyban=remove&ckey=[ckey]'>\[-\]</a><b>[ckey]</b><br />"
+	. += "[ban["message"]] <b><a href='?_src_=holder;stickyban=edit&ckey=[ckey]'>\[Edit\]</a></b><br />"
+	if (ban["admin"])
+		. += "[ban["admin"]]<br />"
+	else
+		. += "LEGACY<br />"
+	. += "Caught keys<br />\n<ol>"
+	for (var/key in ban["keys"])
+		if (ckey(key) == ckey)
+			continue
+		. += "<li><a href='?_src_=holder;stickyban=remove_alt&ckey=[ckey]&alt=[ckey(key)]'>\[-\]</a>[key]</li>"
+	. += "</ol>\n"
+
+/datum/admins/proc/stickyban_show()
+	if(!check_rights(R_BAN))
+		return
+		
+	var/list/bans = world.GetConfig("ban")
+	var/banhtml = ""
+	for(var/key in bans)
```

Comment:
hey! listen!

So I just noticed that this means it's not sorted. I'm gonna put up a pr to sort the output on tg, (just run the bans var thru a sort.)

Just thought you should know.